Transactional session documentation updates and cleanup (#5860)
Particular/docs.particular.net: nservicebus/messaging/batched-dispatch.md

---
title: Batched message dispatch

> summary: Describes how NServiceBus can collect outgoing operations in order to dispatch them more efficiently.

Reviewer: ```suggestion
summary: Describes how NServiceBus collects outgoing operations in order to dispatch them more efficiently.
```

I also like "bundles"

component: Core
reviewed: 2022-09-20
---

> Starting from Versions 6, all outgoing operations executed during message processing (e.g. commands, responses, events) are bundled together and handed to the transport only after the message handling pipeline has completed (the [transactional session](/nservicebus/transactional-session) offers similar behavior for outgoing message operations when the session is committed). This has two benefits:

Reviewer: maybe it needs to be more clear that the transactional session providing similar capabilities outside the pipeline?

```suggestion
Starting from Versions 6, all outgoing operations executed during message processing (e.g. commands, responses, events) are bundled together and handed to the transport only after the message handling pipeline has completed (the [transactional session](/nservicebus/transactional-session) offers similar behavior for `IMessageSession` operations). This has two benefits:
```

Author: Discussed with @tmasternak and @SzymonPobiega we are leaving this one out since mentioning `IMessageSession` with the transactional session could make things confusing

> In Versions 6 and above, all outgoing operations that happen as part of processing a message (e.g. commands, responses, events) will be bundled together and passed to the transport after the message handling pipeline has been completed. With the introduction of the [transactional session](/nservicebus/transactional-session), outgoing message operations can also be bundled together and passed to the transport when the session commits. This has two benefits:

Reviewer 2: ```suggestion
Starting from Versions 6, all outgoing operations executed during message processing (e.g. commands, responses, events) are bundled together and handed to the transport only after the message handling pipeline has completed ([transactional session](/nservicebus/transactional-session) offers similar behavior for outgoing message operations). This has two benefits:
```

Author: I'm unsure about this. I think mentioning the commit is important

>  * Business data will always be committed to storage before any outgoing operations are dispatched. This ensures that there are no outgoing messages in case an exception occurs.

Reviewer 2: ```suggestion
 * Business data operations will be committed to storage before outgoing operations are dispatched. This ensures no outgoing messages are sent in case an exception occurs - during or after the business data operations are committed.
```
